platform/drm/cog-platform-drm.c

```diff
@@ -214,119 +204,83 @@ static struct {
+                g_debug("init_config: atomic modesetting reconfigured to value '%s'",
+                        drm_data.atomic_modesetting ? "true" : "false");
+            }
         }
-    }
 
-    {
-        g_autoptr(GError) lookup_error = NULL;
-        gdouble value = g_key_file_get_double (key_file,
-                                               "drm", "device-scale-factor",
-                                               &lookup_error);
-        if (!lookup_error) {
-            drm_data.device_scale = value;
-            g_debug ("init_config:overriding device_scale value, using %.2f from config",
-                     drm_data.device_scale);
+        {
+            g_autoptr(GError) lookup_error = NULL;
+            gdouble           value = g_key_file_get_double(key_file, "drm", "device-scale-factor", &lookup_error);
+            if (!lookup_error) {
+                drm_data.device_scale = value;
+                g_debug("init_config: overriding device_scale value, using %.2f from config", drm_data.device_scale);
+            }
         }
-    }
-}
 
-
-static void
-destroy_buffer (struct buffer_object *buffer)
-{
-    drmModeRmFB (drm_data.fd, buffer->fb_id);
-    gbm_bo_destroy (buffer->bo);
-
-    if (buffer->export.resource) {
-        wpe_view_backend_exportable_fdo_dispatch_release_buffer (wpe_host_data.exportable,
-                                                                 buffer->export.resource);
-        buffer->export.resource = NULL;
-    }
-#if HAVE_SHM_EXPORTED_BUFFER
-    if (buffer->export.shm_buffer) {
-        wpe_view_backend_exportable_fdo_dispatch_release_shm_exported_buffer (wpe_host_data.exportable,
-                                                                              buffer->export.shm_buffer);
-        buffer->export.shm_buffer = NULL;
+        {
+            g_autofree char *value = g_key_file_get_string(key_file, "drm", "renderer", NULL);
+            if (g_strcmp0(value, "gles") == 0)
+                self->use_gles = true;
+            else if (g_strcmp0(value, "modeset") != 0)
+                self->use_gles = false;
+            else if (value)
+                g_warning("Invalid renderer '%s', using default.", value);
+        }
     }
-#endif
-    g_free (buffer);
-}
 
-static void
-destroy_buffer_notify (struct wl_listener *listener, void *data)
-{
-    struct buffer_object *buffer = wl_container_of (listener, buffer, destroy_listener);
-
-    if (drm_data.committed_buffer == buffer)
-        drm_data.committed_buffer = NULL;
-
-    wl_list_remove (&buffer->link);
-    destroy_buffer (buffer);
-}
-
-static void
-clear_buffers (void)
-{
-    drm_data.committed_buffer = NULL;
-
-    struct buffer_object *buffer, *tmp;
-    wl_list_for_each_safe (buffer, tmp, &drm_data.buffer_list, link) {
-        wl_list_remove (&buffer->link);
-        wl_list_remove (&buffer->destroy_listener.link);
+    if (params_string) {
+        g_auto(GStrv) params = g_strsplit(params_string, ",", 0);
```

Comment:
I guess if you wanted to save some memory here you could use g_strdelimit and transform commas into NUL bytes... But maybe that's being too smart :)

Reply:
I thought of that but it would still need to `g_strdup(params_string)` to have a mutable copy, plus this is done only once during initialization so it is definitely not memory or performance critical. In the end I went with code clarity instead of saving a few allocations.